File: smartbell/src/main/java/com/gymapp/service/CoachRatingService.java
package com.gymapp.service;

import com.gymapp.dto.CoachRatingDTO;
import com.gymapp.dto.RatingRequest;

import java.util.List;

public interface CoachRatingService {

    CoachRatingDTO rateCoach(Long coachId, Long userId, RatingRequest request);

    Double getAverageRating(Long coachId);

    List<CoachRatingDTO> getCoachRatings(Long coachId);
}

File: Frontsmartbell/src/app/services/coach-rating.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { CoachRatingDTO, RatingRequest } from '../models/api.models';

@Injectable({ providedIn: 'root' })
export class CoachRatingService {
  private http = inject(HttpClient);
  private readonly BASE = `${environment.apiUrl}/coaches`;

  rateCoach(coachId: number, request: RatingRequest): Observable<CoachRatingDTO> {
    return this.http.post<CoachRatingDTO>(`${this.BASE}/${coachId}/ratings`, request);
  }

  getAverageRating(coachId: number): Observable<number> {
    return this.http.get<number>(`${this.BASE}/${coachId}/ratings/average`);
  }

  getCoachRatings(coachId: number): Observable<CoachRatingDTO[]> {
    return this.http.get<CoachRatingDTO[]>(`${this.BASE}/${coachId}/ratings`);
  }
}
